Function with multiple return values causes panic when returning an error

The following program `sample.go` triggers a panic:

```go
package main

import (
	"fmt"
)

func foo() ([]string, error) {
	return nil, fmt.Errorf("bar")
}

func main() {
	a, b := foo()
	fmt.Println(a, b)
}
```

Expected result:
```console
$ go run ./sample.go
[] bar
```

Got:
```console
$ yaegi ./sample.go
./sample.go:24:14: panic
./sample.go:28:10: panic
reflect.Set: value of type error is not assignable to type []string
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc000302000, 0x40)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00021be10)
        /Users/berng/go/src/github.com/containous/yaegi/interp/interp.go:320 +0xbc
panic(0x1896980, 0xc000072a40)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0002e40a0, 0xc0002f2200)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:113 +0x20c
panic(0x1896980, 0xc000072a40)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0002e4140, 0xc000187500)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:113 +0x20c
panic(0x1896980, 0xc000072a40)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/runtime/panic.go:969 +0x166
reflect.Value.assignTo(0x18ed3c0, 0xc000042928, 0x94, 0x19c96b8, 0xb, 0x188cac0, 0x0, 0x19bcb20, 0xc00021b901, 0xc00000ed80)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/reflect/value.go:2403 +0x426
reflect.Value.Set(0x188cac0, 0xc00000ed20, 0x197, 0x18ed3c0, 0xc000042928, 0x94)
        /usr/local/Cellar/go/1.14.2_1/libexec/src/reflect/value.go:1532 +0xbd
github.com/containous/yaegi/interp.callBin.func7(0xc0002e4140, 0xc000042800)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:993 +0x228
github.com/containous/yaegi/interp.runCfg(0xc000187500, 0xc0002e4140)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.call.func7(0xc0002e40a0, 0x10b3fcd)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:840 +0xbf1
github.com/containous/yaegi/interp.runCfg(0xc0002f2200, 0xc0002e40a0)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:119 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc000001680, 0xc000187900, 0xc0002e4000)
        /Users/berng/go/src/github.com/containous/yaegi/interp/run.go:97 +0x239
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc000001680, 0xc0002ec000, 0x151, 0x0, 0x0, 0x0, 0x0, 0x0)
        /Users/berng/go/src/github.com/containous/yaegi/interp/interp.go:388 +0x41f
main.main()
        /Users/berng/go/src/github.com/containous/yaegi/cmd/yaegi/yaegi.go:94 +0x548
```

--- a/_test/fun19.go
+++ b/_test/fun19.go
@@ -1,0 +1,17 @@
+package main
+
+import (
+	"fmt"
+)
+
+func foo() ([]string, error) {
+	return nil, fmt.Errorf("bar")
+}
+
+func main() {
+	a, b := foo()
+	fmt.Println(a, b)
+}
+
+// Output:
+// [] bar

--- a/_test/fun20.go
+++ b/_test/fun20.go
@@ -1,0 +1,19 @@
+package main
+
+import "fmt"
+
+var myerr error = fmt.Errorf("bar")
+
+func ferr() error { return myerr }
+
+func foo() ([]string, error) {
+	return nil, ferr()
+}
+
+func main() {
+	a, b := foo()
+	fmt.Println(a, b)
+}
+
+// Output:
+// [] bar

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -823,9 +823,13 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 						}
 					} else {
 						n.typ = &itype{cat: valueT, rtype: typ.Out(0)}
-						n.findex = sc.add(n.typ)
-						for i := 1; i < typ.NumOut(); i++ {
-							sc.add(&itype{cat: valueT, rtype: typ.Out(i)})
+						if n.anc.kind == returnStmt {
+							n.findex = childPos(n)
+						} else {
+							n.findex = sc.add(n.typ)
+							for i := 1; i < typ.NumOut(); i++ {
+								sc.add(&itype{cat: valueT, rtype: typ.Out(i)})
+							}
 						}
 					}
 				}
@@ -836,9 +840,13 @@ func (interp *Interpreter) cfg(root *node, pkgID string) ([]*node, error) {
 				}
 				if typ := n.child[0].typ; len(typ.ret) > 0 {
 					n.typ = typ.ret[0]
-					n.findex = sc.add(n.typ)
-					for _, t := range typ.ret[1:] {
-						sc.add(t)
+					if n.anc.kind == returnStmt {
+						n.findex = childPos(n)
+					} else {
+						n.findex = sc.add(n.typ)
+						for _, t := range typ.ret[1:] {
+							sc.add(t)
+						}
 					}
 				} else {
 					n.findex = -1

--- a/interp/run.go
+++ b/interp/run.go
@@ -687,7 +687,7 @@ func call(n *node) {
 	case returnStmt:
 		// Function call from a return statement: forward return values (always at frame start).
 		for i := range rtypes {
-			j := i
+			j := n.findex + i
 			ret := n.child[0].typ.ret[i]
 			callret := n.anc.val.(*node).typ.ret[i]
 			if callret.cat == interfaceT && ret.cat != interfaceT {
@@ -983,6 +983,7 @@ func callBin(n *node) {
 		case aReturn:
 			// The function call is part of a return statement, store output results
 			// directly in the frame location of outputs of the current function.
+			b := childPos(n)
 			n.exec = func(f *frame) bltn {
 				in := make([]reflect.Value, l)
 				for i, v := range values {
@@ -990,7 +991,7 @@ func callBin(n *node) {
 				}
 				out := value(f).Call(in)
 				for i, v := range out {
-					f.data[i].Set(v)
+					f.data[b+i].Set(v)
 				}
 				return tnext
 			}

